app/code/Magento/InventorySalesFrontendUi/Plugin/Block/Stockqty/AbstractStockqtyPlugin.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySalesFrontendUi\Plugin\Block\Stockqty;
+
+use Magento\InventoryConfigurationApi\Api\Data\StockItemConfigurationInterface;
+use Magento\InventoryConfigurationApi\Api\GetStockItemConfigurationInterface;
+use Magento\InventorySalesApi\Api\GetProductSalableQtyInterface;
+use Magento\InventorySalesApi\Model\StockByWebsiteIdResolverInterface;
+
+class AbstractStockqtyPlugin
+{
+    /**
+     * @var GetStockItemConfigurationInterface
+     */
+    private $getStockItemConfiguration;
+
+    /**
+     * @var StockByWebsiteIdResolverInterface
+     */
+    private $stockByWebsiteId;
+
+    /**
+     * @var GetProductSalableQtyInterface
+     */
+    private $getProductSalableQty;
+
+    /**
+     * @param StockByWebsiteIdResolverInterface $stockByWebsiteId
+     * @param GetStockItemConfigurationInterface $getStockItemConfig
+     * @param GetProductSalableQtyInterface $getProductSalableQty
+     */
+    public function __construct(
+        StockByWebsiteIdResolverInterface $stockByWebsiteId,
+        GetStockItemConfigurationInterface $getStockItemConfig,
+        GetProductSalableQtyInterface $getProductSalableQty
+    ) {
+        $this->getStockItemConfiguration = $getStockItemConfig;
+        $this->stockByWebsiteId = $stockByWebsiteId;
+        $this->getProductSalableQty = $getProductSalableQty;
+    }
+
+    /**
+     * @param \Magento\CatalogInventory\Block\Stockqty\AbstractStockqty $subject
+     * @param callable $proceed
+     * @return bool
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function aroundIsMsgVisible(
+        \Magento\CatalogInventory\Block\Stockqty\AbstractStockqty $subject,
+        callable $proceed
+    ): bool {
+        $sku = $subject->getProduct()->getSku();
+        $websiteId = (int)$subject->getProduct()->getStore()->getWebsiteId();
+        $stockId = (int)$this->stockByWebsiteId->execute($websiteId)->getStockId();
+        $stockItemConfig = $this->getStockItemConfiguration->execute($sku, $stockId);
+        if (null === $stockItemConfig) {
+            return false;
+        }
+        return $stockItemConfig->getBackorders() === StockItemConfigurationInterface::BACKORDERS_NO
+            && $this->getProductSalableQty->execute($sku, $stockId) > 0
```

Comment:
we don't need this check 
` && $this->getProductSalableQty->execute($sku, $stockId) > 0`

because in this case product would not be salable at all